src/Tree/AbstractNode.php

```diff
@@ -65,10 +65,12 @@ public function getParent(): ?AbstractNode
 		return $this->parent;
 	}
 
-	public function setParent(?AbstractNode $parent): void
+	public function setParent(?ParentNode $closestParent): void
```

Comment:
Why have you changed it?

Reply:
`parent` property have typehint ParentNode. And only ParentNode can have childs, so it seemed like a valid solution